test/blackbox-tests/deriving-separate.t

```diff
@@ -0,0 +1,108 @@
+Tests for deriving `make_opt_keys, getters_setters`
+
+  $ . ./setup.sh
+
+`[@@deriving make_opt_keys]` just derives the constructor
+
+  $ cat > x.ml <<EOF
+  > type chartDataItemType =
+  >   { height : int
+  >   ; foo : string [@mel.as "width"]
+  >   } [@@deriving make_opt_keys]
+  > 
+  > let t = chartDataItemType ~height:2 ~foo:"bar"
+  > EOF
+  $ melc -ppx melppx -dsource x.ml
+  type chartDataItemType = {
+    height: int ;
+    foo: string [@mel.as "width"]}[@@deriving make_opt_keys]
+  include
+    struct
+      let _ = fun (_ : chartDataItemType) -> ()
+      external chartDataItemType :
+        height:int -> foo:string -> chartDataItemType = ""
+          "\132\149\166\190\000\000\000\023\000\000\000\t\000\000\000\024\000\000\000\022\145\160\160A\144&height\160\160A\144%width@"
+      [@@ocaml.warning "-unboxable-type-in-prim-decl"][@@ocaml.warning
+                                                        "-unboxable-type-in-prim-decl"]
+    end[@@ocaml.doc "@inline"][@@merlin.hide ]
+  let t = ((chartDataItemType ~height:2 ~foo:"bar")
+    [@ocaml.warning "-ignored-extra-argument"][@ocaml.warning
+                                                "-ignored-extra-argument"])
+  // Generated by Melange
+  'use strict';
+  
+  
+  var t = {
+    height: 2,
+    width: "bar"
+  };
+  
+  exports.t = t;
+  /* No side effect */
+
+`[@@deriving getters_setters]` just derives the getters / setters
+
+  $ cat > x.ml <<EOF
+  > type chartDataItemType =
+  >   { height : int
+  >   ; foo : string [@mel.as "width"]
+  >   } [@@deriving getters_setters]
+  > 
+  > EOF
+  $ melc -ppx melppx -dsource x.ml
+  type chartDataItemType = {
+    height: int ;
+    foo: string [@mel.as "width"]}[@@deriving getters_setters]
+  include
+    struct
+      let _ = fun (_ : chartDataItemType) -> ()
+      external heightGet :
+        chartDataItemType -> int = ""
+          "\132\149\166\190\000\000\000\r\000\000\000\004\000\000\000\012\000\000\000\011\176\145AA\168&height@"
+      [@@ocaml.warning "-unboxable-type-in-prim-decl"][@@internal.arity 1]
+      [@@ocaml.warning "-unboxable-type-in-prim-decl"]
```

Comment:
Just curious, why is this warning silencing needed?

Reply:
here's one example where you reported the warning existing when we generate externals: https://github.com/melange-re/melange/pull/415